assets/src/BlockEditorValidation.js

```diff
@@ -26,19 +48,38 @@ const PrePublishCheckList = () => {
   const invalidElements = elements.filter( element => !isValid( element ) );
   elements.forEach( element => { element.classList.remove( 'cmb2_required_field_error') } );
   invalidElements.forEach( element => { element.classList.toggle( 'cmb2_required_field_error') } );
-  const postIsValid = invalidElements.length > 0
+  const metaInvalid = invalidElements.length > 0
 
-  if ( postIsValid ) {
+  if ( metaInvalid ) {
     // Open "Analytics & Tracking" fields metabox, if closed.
     $('#p4_campaign_fields').removeClass('closed');
     checkListMsg.push( __( 'Please check "Analytics & Tracking" section for required fields.', 'planet4-master-theme-backend' ) );
     invalidElements.forEach( element => { checkListMsg.push( ' - ' + $(element).parent().find('label').text() + ' is a required field' ) } );
   }
 
-  let classname = '';
-  if ( postIsValid ) {
+  const blocks = select( 'core/block-editor' ).getBlocks();
+  const blockResults = blocks.map( ( block ) => {
+    const validations = blockValidations[ block.name ];
+    if ( !validations ) {
+      return {block, invalids: []};
+    }
+    const results = Object.keys( validations ).map( attrName => {
+      const validate = validations[ attrName ];
+
+      return { attr: attrName, ...validate( block.attributes[ attrName ] ) };
+    } );
+    const invalids = results.filter( result => !result.isValid );
+
+    return { block, invalids };
+  } );
+
+  const invalidBlocks = blockResults.filter( block => block.invalids.length > 0 );
+
+    let classname = '';
+  if ( metaInvalid || invalidBlocks.length > 0) {
     dispatch( 'core/editor' ).lockPostSaving();
     classname = 'p4-plugin-pre-publish-panel-error';
+    invalidBlocks.forEach( block => block.invalids.forEach( invalid => checkListMsg.push( ...invalid.messages ) ) );
```

Comment:
The block name was added in the message. I moved it to the front of the message to make it more clear. Adding it to the BlockEditorValidator would be better as we could show it as a 'heading' to the errors. But it would have the same styling as the normal errors so I don't think that'll work.